### ISSUE TITLE
Geth installation using docker

### ISSUE THREAD

**Issue by BardoBravo** [ethereum/go-ethereum#14479]
#### System information

Geth version: 1.5.9
OS & Version: Windows

I have installed the Ethereum blockchain using docker (testnet), and know I would like to play with contracts using solidity. I am able to create accounts in my docker instance by using command get account new, but how can I access them now?

Regards,
Cláudio Ramos

**Comment by Ali92hm** [comment]
It is not recommended to keep data in Docker containers. So if you were keeping the data in the container you will need to do the following do get it back:

`docker cp <containerID>:/root/.ethereum/keystore DEST_PATH`

**Comment by BardoBravo** [comment]
Hello @Ali92hm ,
Although the installation was done with Docker, the blockchain data is locally in my PC, so I thought I could create accounts and expose them with geth --rpc commands. Is this possible?
Hugs

**Comment by Ali92hm** [comment]
Oh you mean use the RPC methods?

When you run the docker container you need to expose the port 8545 and enable RPC. Try the following:

`docker run -it <your other flags> -p 8545:8545 ethereum/client-go <your geth flags> --rpc`

**Comment by BardoBravo** [comment]
Well, in order to install the ethereum blockchain I sued the command:
docker run -d --name ethereum-node -v /Users/alice/ethereum:/root \
           -p 8545:8545 -p 30303:30303 \
           ethereum/client-go --fast --cache=512

It seems to be already ready to enable port 8545... If I do the same --prc command in shell, the localhost:8545 is enabled, but throws a JSON error (error -32600 ).

**Comment by BardoBravo** [comment]
@Ali92hm

**Comment by Ali92hm** [comment]
Try this:

`docker run -d --name ethereum-node -v /Users/alice/ethereum:/root -p 8545:8545 -p 30303:30303 ethereum/client-go --fast --cache=512 --rpc --rpcaddr 0.0.0.0`

And then to test it run `curl -X POST --data '{"jsonrpc":"2.0","method":"net_peerCount","params":[],"id":74}' localhost:8545`

**Comment by Ali92hm** [comment]
@BardoBravo Did that solve your problem?

**Comment by BardoBravo** [comment]
Hello @Ali92hm ,

Once I execute the cURL command, I get the following error message:

> {"jsonrpc":"2.0","error":{"code":-32600,"message":"invalid character '\ \ '' looking for beginning of value"}}

Any idea?

**Comment by Ali92hm** [comment]
That might be a windows shell problem. Does that happen when you try to use `Web3.js` as well?

**Comment by BardoBravo** [comment]
@Ali92hm how would I do  so?

**Comment by BardoBravo** [comment]
@Ali92hm From this [post](https://ethereum.stackexchange.com/questions/9938/unable-to-connect-ethereum-node-even-rpc-port-8545-is-open) it seems that the EOF return from localhost:8545 is expected, thus we should use truffle to develop apps on private network. Do you agree?

**Comment by Ali92hm** [comment]
No, there is no "should". You can use truffle. This problems definitely originates from something else.

I made this quick test try running it: [https://github.com/Ali92hm/web3-test](https://github.com/Ali92hm/web3-test)

**Comment by BardoBravo** [comment]
@Ali92hm 
I called geth --testnet --rpc --rpcaddr 0.0.0.0 --rpccorsdomain 0.0.0.0 console and opened other shell interface, In it, I followed your instruction and the result was as you said:

Web3 connected to RPC
Web3 listening:  true
Web3 peerCount:  0
Web3 syncing:  false
Web3 blockNumber:  208

localhost:8545 still shows the same JSON error.

Any conclusion from that?

**Comment by Ali92hm** [comment]
Yeah that means that your RPC works, and the errors are probably because of the wrong things that are being passed to the endpoint. I don't know why your geth is not syncing though.

[Web3js](https://github.com/ethereum/wiki/wiki/JavaScript-API) is one of the way of interaction with `geth` if you want to write JavaScript. Otherwise there are wrappers in any other language!

**Comment by BardoBravo** [comment]
@Ali92hm is there a way I can track what is being passed to the endpoint?

**Comment by Ali92hm** [comment]
You should be able to use Wireshark to monitor the traffic. 
But let me as you a question, how are you planning to write your application? what programming language are you trying to use? What are you trying to make?

I'm asking these questions cause, the majority of apps use `Web3js` to interact with `Geth` and that seems to be working for you!

**Comment by Ali92hm** [comment]
@BardoBravo any luck?
